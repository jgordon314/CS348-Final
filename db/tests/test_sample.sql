-- These sample queries are for the basic features outlined in milestone 2. Between milestone 2 and milestone 3 some basic and advanced features were switched around, so the numbering of test_sample.sql may not correspond to test_production.sql.

-- Load sample fixtures
SOURCE db/fixtures/load_sample.sql;

-- R6
SELECT id, username FROM Users WHERE username = 'that_yuck' AND password = 'Amogus32!';
INSERT INTO Users (username, password) VALUES ('billnye', 'the5c1enceguy!');
SELECT id, username FROM Users WHERE username = 'billnye' AND password = 'the5c1enceguy!';

-- R7
SELECT * FROM Characters WHERE user_id = 2; 

-- R8
START TRANSACTION; 

INSERT INTO SkillDeltas (
  armor_class, current_health, max_health,
  strength, dexterity, constitution, intelligence, wisdom, charisma,
  acrobatics, animal_handling, arcana, athletics, deception,
  history, insight, intimidation, investigation, medicine,
  nature, perception, performance, persuasion, religion,
  sleight_of_hand, stealth, survival
) VALUES (
  3, 12, 20,
  10, 14, 12, 11, 13, 9,
  2, 1, 3, 4, 0,
  2, 3, 1, 2, 1,
  2, 4, 0, 1, 3,
  2, 5, 2
);

INSERT INTO Spells (name, level, description, skill_delta_id, components) VALUES ('Firebolt', 0, 'A bolt of fire deals 1d10 damage.', LAST_INSERT_ID(), '');

COMMIT;

SELECT name, level, description FROM Spells WHERE id = LAST_INSERT_ID();


START TRANSACTION; 

INSERT INTO SkillDeltas (
  armor_class, current_health, max_health,
  strength, dexterity, constitution, intelligence, wisdom, charisma,
  acrobatics, animal_handling, arcana, athletics, deception,
  history, insight, intimidation, investigation, medicine,
  nature, perception, performance, persuasion, religion,
  sleight_of_hand, stealth, survival
) VALUES (
  2, 8, 15,
  9, 13, 11, 12, 10, 8,
  1, 0, 4, 2, 3,
  1, 2, 0, 3, 1,
  2, 3, 1, 2, 1,
  2, 3, 2
);

INSERT INTO Abilities (name, type, description, skill_delta_id) VALUES ('Heal', 'action', 'Restore 2d4 + 2 HP', LAST_INSERT_ID());

INSERT INTO Items (name, description, ability_id) VALUES ('Healing Potion', 'Effect: Restore 2d4 + 2 HP', LAST_INSERT_ID());

COMMIT;

SELECT
Items.name AS item_name, Items.description AS item_description,
Abilities.name AS ability_name, Abilities.type AS ability_type, Abilities.description AS ability_description
FROM Items, Abilities WHERE Items.id = LAST_INSERT_ID() AND Items.id = Abilities.id;


START TRANSACTION; 

INSERT INTO SkillDeltas (
  armor_class, current_health, max_health,
  Strength, Dexterity, Constitution, Intelligence, Wisdom, Charisma,
  Acrobatics, Animal_Handling, Arcana, Athletics, Deception,
  History, Insight, Intimidation, Investigation, Medicine,
  Nature, Perception, Performance, Persuasion, Religion,
  Sleight_of_Hand, Stealth, Survival
) VALUES (
  5, 20, 25,
  15, 12, 13, 11, 10, 14,
  3, 2, 1, 5, 2,
  4, 1, 2, 3, 2,
  1, 3, 2, 3, 4,
  1, 1, 3
);

INSERT INTO Abilities (name, type, description, skill_delta_id) VALUES ('Rage', 2, 'Enter a frenzied rage for 1 minute.', LAST_INSERT_ID());

COMMIT;

SELECT name, type, description FROM Abilities WHERE id = LAST_INSERT_ID();

-- R9
CREATE VIEW CharacterBaseCharismas AS
SELECT Characters.id AS character_id, Characters.name AS character_name, SkillDeltas.charisma AS character_base_charisma
FROM Characters, SkillDeltas WHERE Characters.base_stat_id = SkillDeltas.id;

SELECT * FROM CharacterBaseCharismas WHERE character_id = 4;
UPDATE SkillDeltas SET charisma = charisma + 3 WHERE id = (SELECT base_stat_id FROM Characters WHERE id = 4);
SELECT * FROM CharacterBaseCharismas WHERE character_id = 4;

DROP VIEW CharacterBaseCharismas;

-- R10
INSERT INTO CharacterSpellList (character_id, spell_id, activations) VALUES (2, 3, 3);
SELECT Characters.name AS charcter_name, Spells.name AS spell_name, CharacterSpellList.activations AS activations
FROM Characters, Spells, CharacterSpellList
WHERE Characters.id = 2 AND Characters.id = CharacterSpellList.character_id AND Spells.id = CharacterSpellList.spell_id;

INSERT INTO CharacterInventory (character_id, item_id, quantity) VALUES (3, 6, 2);
SELECT Characters.name AS charcter_name, Items.name AS item_name, CharacterInventory.quantity AS item_quantity
FROM Characters, Items, CharacterInventory
WHERE Characters.id = 3 AND Characters.id = CharacterInventory.character_id AND Items.id = CharacterInventory.item_id;

INSERT INTO CharacterAbilities (character_id, ability_id) VALUES (4, 9);
SELECT Characters.name AS charcter_name, Abilities.name AS ability_name
FROM Characters, Abilities, CharacterAbilities
WHERE Characters.id = 4 AND Characters.id = CharacterAbilities.character_id AND Abilities.id = CharacterAbilities.ability_id;
