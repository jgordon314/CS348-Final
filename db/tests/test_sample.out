Query OK, 0 rows affected (0.068 sec)

Query OK, 0 rows affected (0.020 sec)

Query OK, 0 rows affected (0.014 sec)

Query OK, 0 rows affected (0.020 sec)

Query OK, 0 rows affected (0.012 sec)
Records: 0  Duplicates: 0  Warnings: 0

Query OK, 0 rows affected (0.019 sec)

Query OK, 0 rows affected (0.014 sec)

Query OK, 0 rows affected (0.013 sec)

Query OK, 0 rows affected (0.016 sec)

Query OK, 0 rows affected (0.016 sec)

Query OK, 0 rows affected (0.014 sec)

Query OK, 6 rows affected (0.002 sec)
Records: 6  Duplicates: 0  Warnings: 0

Query OK, 12 rows affected (0.003 sec)
Records: 12  Duplicates: 0  Warnings: 0

Query OK, 4 rows affected (0.004 sec)
Records: 4  Duplicates: 0  Warnings: 0

Query OK, 13 rows affected (0.004 sec)
Records: 13  Duplicates: 0  Warnings: 0

Query OK, 9 rows affected (0.003 sec)
Records: 9  Duplicates: 0  Warnings: 0

Query OK, 8 rows affected (0.002 sec)
Records: 8  Duplicates: 0  Warnings: 0

Query OK, 12 rows affected (0.003 sec)
Records: 12  Duplicates: 0  Warnings: 0

Query OK, 11 rows affected (0.005 sec)
Records: 11  Duplicates: 0  Warnings: 0

Query OK, 14 rows affected (0.003 sec)
Records: 14  Duplicates: 0  Warnings: 0

+----+-----------+
| id | username  |
+----+-----------+
|  1 | that_yuck |
+----+-----------+
1 row in set (0.000 sec)

Query OK, 1 row affected (0.002 sec)

+----+----------+
| id | username |
+----+----------+
|  7 | billnye  |
+----+----------+
1 row in set (0.000 sec)

+----+-------------+-----------------------------------------+---------+--------------+
| id | name        | description                             | user_id | base_stat_id |
+----+-------------+-----------------------------------------+---------+--------------+
|  1 | Damian Skye | Male Half-Elf, Blood Hunter 3/ Wizard 1 |       2 |            1 |
+----+-------------+-----------------------------------------+---------+--------------+
1 row in set (0.001 sec)

Query OK, 0 rows affected (0.000 sec)

Query OK, 1 row affected (0.001 sec)

Query OK, 1 row affected (0.001 sec)

Query OK, 0 rows affected (0.002 sec)

+----------+-------+-----------------------------------+
| name     | level | description                       |
+----------+-------+-----------------------------------+
| Firebolt |     0 | A bolt of fire deals 1d10 damage. |
+----------+-------+-----------------------------------+
1 row in set (0.000 sec)

Query OK, 0 rows affected (0.000 sec)

Query OK, 1 row affected (0.001 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 0 rows affected (0.003 sec)

+----------------+----------------------------+-----------------+--------------+---------------------+
| item_name      | item_description           | ability_name    | ability_type | ability_description |
+----------------+----------------------------+-----------------+--------------+---------------------+
| Healing Potion | Effect: Restore 2d4 + 2 HP | Attack (Dagger) | action       | 1d4 Slashing        |
+----------------+----------------------------+-----------------+--------------+---------------------+
1 row in set (0.000 sec)

Query OK, 0 rows affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 1 row affected (0.000 sec)

Query OK, 0 rows affected (0.002 sec)

+------+--------+-------------------------------------+
| name | type   | description                         |
+------+--------+-------------------------------------+
| Rage | action | Enter a frenzied rage for 1 minute. |
+------+--------+-------------------------------------+
1 row in set (0.000 sec)

Query OK, 0 rows affected (0.003 sec)

+--------------+----------------+-------------------------+
| character_id | character_name | character_base_charisma |
+--------------+----------------+-------------------------+
|            4 | Jill Rogue     |                      16 |
+--------------+----------------+-------------------------+
1 row in set (0.001 sec)

Query OK, 1 row affected (0.002 sec)
Rows matched: 1  Changed: 1  Warnings: 0

+--------------+----------------+-------------------------+
| character_id | character_name | character_base_charisma |
+--------------+----------------+-------------------------+
|            4 | Jill Rogue     |                      19 |
+--------------+----------------+-------------------------+
1 row in set (0.001 sec)

Query OK, 0 rows affected (0.003 sec)

Query OK, 1 row affected (0.002 sec)

+---------------+------------------+-------------+
| charcter_name | spell_name       | activations |
+---------------+------------------+-------------+
| Krovav        | Friends          |           0 |
| Krovav        | Longstrider      |           3 |
| Krovav        | Toll the Dead    |           0 |
| Krovav        | Disguise Self    |           0 |
| Krovav        | Eldritch Blast   |           0 |
| Krovav        | Armor of Agathys |           0 |
| Krovav        | Fly              |           1 |
+---------------+------------------+-------------+
7 rows in set (0.001 sec)

Query OK, 1 row affected (0.002 sec)

+----------------+-------------------+---------------+
| charcter_name  | item_name         | item_quantity |
+----------------+-------------------+---------------+
| John Barbarian | Rations (1 day)   |             5 |
| John Barbarian | Potion of Healing |             2 |
| John Barbarian | Chain Shirt       |             2 |
+----------------+-------------------+---------------+
3 rows in set (0.001 sec)

Query OK, 1 row affected (0.002 sec)

+---------------+---------------------+
| charcter_name | ability_name        |
+---------------+---------------------+
| Jill Rogue    | Unarmed Strike      |
| Jill Rogue    | Dash                |
| Jill Rogue    | Dash (Bonus Action) |
| Jill Rogue    | Attack (Dagger)     |
+---------------+---------------------+
4 rows in set (0.001 sec)
